base/uk.ac.stfc.isis.ibex.e4.ui.perspectiveswitcher/src/uk/ac/stfc/isis/ibex/e4/ui/perspectiveswitcher/views/PerspectiveSwitcherView.java

```diff
@@ -96,11 +142,32 @@ public void handleEvent(Event event) {
 	}
 
 	private void addSeparator(Composite parent) {
-		Label separator = new Label(parent, SWT.HORIZONTAL | SWT.SEPARATOR);
-		separator.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		separator = new Label(parent, SWT.HORIZONTAL | SWT.SEPARATOR);
+		separatorGD = new GridData(SWT.FILL, SWT.FILL, true, false);
+		separator.setLayoutData(separatorGD);
 	}
 
 	private void addResetCurrentPerspectiveShortcut(Composite parent) {
-		new ResetLayoutButton(parent, perspectivesProvider);
+	    ResetLayoutButtonViewModel model = new ResetLayoutButtonViewModel();
+	    model.addPropertyChangeListener("triggerReset", new PropertyChangeListener() {
+            @Override
+            public void propertyChange(PropertyChangeEvent evt) {
+                maximise();
+            }
+        });
+	    buttonModels.add(model);
+	    new ResetLayoutButton(parent, perspectivesProvider, model);
```

Comment:
You can add a listener on this button here rather than on the model then you don't need to pass the resetTrigger event around.